remote/internal/metrics.go

```diff
@@ -33,38 +43,42 @@ type PendulumEntry struct {
 //
 // Returns:
 // - A slice of PendulumMetric structs containing the aggregated metrics.
-func AggregatePendulumMetrics(data [][]string, timeout_len float64, rangeType string) []PendulumMetric {
-	// get number of columns
-	n := len(data[0])
-
+func AggregatePendulumMetrics(
+	data [][]string,
+	timeout_len float64,
+	rangeType string,
+	sections []interface{}) []PendulumMetric {
 	// create waitgroup
 	var wg sync.WaitGroup
 
 	// create buffered channel to store results and avoid deadlock in main
-	res := make(chan PendulumMetric, n-2)
+	res := make(chan PendulumMetric, len(sections))
 
-	// iterate through each metric column (except for 'active' and 'time') and create goroutine for each
-	for m := 1; m < n-1; m++ {
+	// iterate through each metric column as specified in Sections config
+	// and create goroutine for each
+	for _, colName := range sections {
 		wg.Add(1)
 		go func(m int) {
 			defer wg.Done()
 			aggregatePendulumMetric(data, m, timeout_len, rangeType, res)
-		}(m)
+		}(csvColumns[colName.(string)])
```

Comment:
We need to add a check for if the input column name is actually in the csvColumns map, otherwise the output defaults to 0 and a report is created for the 'active' column. (This happens if one of the 'sections' table inputs is invalid.)

```suggestion
		m, ok := csvColumns[colName.(string)]
		if !ok {
			continue
		}

		wg.Add(1)
		go func(m int) {
			defer wg.Done()
			aggregatePendulumMetric(data, m, timeout_len, rangeType, res)
		}(m)
```

Reply:
That is exactly what happens. I did not test out this case.

Reply:
This is also taken care of in the latest commits